pkg/provider/acme/provider.go

```diff
@@ -370,22 +371,33 @@ func (p *Provider) getClient() (*lego.Client, error) {
 	return p.client, nil
 }
 
+// Maximum allowed value for Configuration.ClientTimeout.
+// We multiply the timeout by 4, so pick `"time".maxDuration / 4`.
+const maxClientTimeout = time.Duration((1<<63 - 1) / 4)
+
 func (p *Provider) createHTTPClient() (*http.Client, error) {
 	tlsConfig, err := p.createClientTLSConfig()
 	if err != nil {
 		return nil, fmt.Errorf("creating client TLS config: %w", err)
 	}
 
+	clientTimeout := 30 * time.Second
+	if p.Configuration != nil && p.Configuration.ClientTimeout != nil {
+		timeout := time.Duration(*p.Configuration.ClientTimeout)
+		// clamp value to avoid overflows
+		clientTimeout = max(min(timeout, maxClientTimeout), 0)
+	}
+
 	return &http.Client{
-		Timeout: 2 * time.Minute,
+		Timeout: 4 * clientTimeout,
 		Transport: &http.Transport{
 			Proxy: http.ProxyFromEnvironment,
 			DialContext: (&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
+				Timeout:   clientTimeout,
+				KeepAlive: clientTimeout,
 			}).DialContext,
-			TLSHandshakeTimeout:   30 * time.Second,
-			ResponseHeaderTimeout: 30 * time.Second,
+			TLSHandshakeTimeout:   clientTimeout,
+			ResponseHeaderTimeout: clientTimeout,
```

Comment:
Hi @tkw1536,

Thanks this contribution!

At a glance, having an option defining all timeouts at once is not something we would do.
It seems that only the `ResponseHeaderTimeout` timeout could be expose to solve the described use case.
WDYT of iterating in that way?

Reply:
Only setting `client.Transport.ResponseHeaderTimeout` should address the issue. 
However it does not make sense to have `client.Timeout < client.Transport.ResponseHeaderTimeout`, so we would at least have to set that field also. 

I've updated the PR (and rebased onto master).

Reply:
I have two more questions:

- Any opinions on the name of the configuration option? It is currently `clientTimeout` (client as in ACME client). 
- I'm currently using a `*ptypes.Duration`. The "default" of 30s is applied only in `createHTTPClient` (when it is nil). Does it make sense to use a non-pointer `ptypes.Duration` with a default in `SetDefaults` instead?
I would imagine that would need quite a few changes in tests all over the place (wherever a config is loaded and SetDefaults is called).

Reply:
@tkw1536 I agree, I think I overlooked the problematic here.

I think we should expose both Timeout and ResponseHeaderTimeout.
A check should ensure that ResponseHeaderTimeout is not greater than Timeout, and this should be a requirement for Traefik to startup.

> I'm currently using a *ptypes.Duration. The "default" of 30s is applied only in createHTTPClient (when it is nil). Does it make sense to use a non-pointer ptypes.Duration with a default in SetDefaults instead?
I would imagine that would need quite a few changes in tests all over the place (wherever a config is loaded and SetDefaults is called).

Yes, there is no need for a pointer, as the `SetDefaults` would always replace the nil value.

Reply:
I've updated to use non-pointers and introduced a second ResponseHeaderTimeout timeout. 

> A check should ensure that ResponseHeaderTimeout is not greater than Timeout, and this should be a requirement for Traefik to startup.

I've introduced a check inside of the `Init` method. 
When setting values incorrectly the provider fails to start:

```
ERR The ACME resolve is skipped from the resolvers list error="clientTimeout must be at least clientResponseHeaderTimeout" resolver=institution
ERR error="HTTP challenge is not enabled" entryPointName=web routerName=acme-http@internal
```
The rest operates normally, only the provider will not start. 
Is there a different place you were thinking of putting the check to preventing traefik startup completely?

Reply:
Or not, need to rebase first.

Reply:
Ok, rebase done.

Reply:
Marking as resolved, as this was done in https://github.com/traefik/traefik/pull/11637/commits/47f2a3d534717e0100852ede02d654663195e278.